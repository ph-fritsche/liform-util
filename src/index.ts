export { isShallowEqual } from './comparison/isShallowEqual'
export { intervalFromString, Interval } from './date/interval'
export { getIndicesOfDescendant } from './dom/indicesOfDescendant'
export { useBoundFunction } from './react/boundFunction'
export { useForkedCallback } from './react/forkedCallback'
export { useForkedRef, updateRef } from './react/forkedRef'
export { useId } from './react/id'
export { ComponentForwardRef, WrapperProps, ForwardRefWrapper, createForwardRefWrapper } from './react/wrapper'
export { isEscaped, countUnescaped, hasUnescaped, indexOfUnescaped } from './string/escaped'
export { filterObject } from './object/filterObject'
