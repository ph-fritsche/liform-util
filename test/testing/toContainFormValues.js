import '../../src/testing'

const body = document.createElement('body')
body.innerHTML = `
    <form id='testform'>
        <input type='text' name='foo' value='fooValue'/>
        <input type='text' name='bar' value='barValue'/>
    </form>
`

const form = body.querySelector('#testform')

it('Pass for equal values', () => {
    expect(() => {
        expect(form).toContainFormValues({
            'foo': 'fooValue',
            'bar': 'barValue',
        })
    }).not.toThrowError()
})

it('Fail for missing field', () => {
    expect(() => {
        expect(form).toContainFormValues({
            'foo': 'fooValue',
            'bar': 'barValue',
            'baz': 'bazValue',
        })
    }).toThrowErrorMatchingSnapshot()
})

it('Fail for wrong value', () => {
    expect(() => {
        expect(form).toContainFormValues({
            'foo': 'fooValue',
            'bar': 'bazValue',
        })
    }).toThrowErrorMatchingSnapshot()
})

it('Pass for extra field', () => {
    expect(() => {
        expect(form).toContainFormValues({
            'foo': 'fooValue',
        })
    }).not.toThrowError()
})
