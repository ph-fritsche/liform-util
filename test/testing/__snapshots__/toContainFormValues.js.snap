// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fail for missing field 1`] = `
"expect(FORM).toContainFormValues({
  \\"foo\\": \\"fooValue\\",
  \\"bar\\": \\"barValue\\",
  \\"baz\\": \\"bazValue\\"
})

- Expected  - 1
+ Received  + 0

  Object {
    \\"bar\\": \\"barValue\\",
-   \\"baz\\": \\"bazValue\\",
    \\"foo\\": \\"fooValue\\",
  }"
`;

exports[`Fail for wrong value 1`] = `
"expect(FORM).toContainFormValues({
  \\"foo\\": \\"fooValue\\",
  \\"bar\\": \\"bazValue\\"
})

- Expected  - 1
+ Received  + 1

  Object {
-   \\"bar\\": \\"bazValue\\",
+   \\"bar\\": \\"barValue\\",
    \\"foo\\": \\"fooValue\\",
  }"
`;
